src/reducers/AddEditSiteReducer.js

```diff
@@ -165,36 +211,41 @@ export default (state = INITIAL_STATE, action) => {
                 readyLongitude: longitude
             };
 
-        case ADDSITE_IMAGE_UPDATED:
+        case `${prefix}_${ADDSITE_IMAGE_UPDATED}`:
             const {updatedImage} = payload;
             const siteImageData = updatedImage.hasImage ? updatedImage.base64 : '';
 
             return {...state, siteImageData};
 
-        case SITE_DETAIL_CHECKBOX_UPDATED:
+        case `${prefix}_${SITE_DETAIL_CHECKBOX_UPDATED}`:
             const {siteDetailCheckboxKey} = payload;
             const updatedSiteDetailCheckboxesKeys = updateSiteDetailCheckboxesKeys(state, siteDetailCheckboxKey);
 
             return {...state, siteDetailCheckboxesKeys: updatedSiteDetailCheckboxesKeys};
 
-        case CHECK_IF_SITE_IS_READY:
+        case `${prefix}_${CHECK_IF_SITE_IS_READY}`:
             const siteReadyForUpload = siteIsReadyForUpload(state);
 
             return {...state, siteReadyForUpload};
 
-        case ADD_SITE_SUCCESS:
+        case `${prefix}_${ADD_SITE_SUCCESS}`:
             return {...INITIAL_STATE};
 
-        case ADD_SITE_FAILURE:
+        case `${prefix}_${ADD_SITE_FAILURE}`:
             return {...state};
 
-        case MAP_IS_INITIALIZING:
+        case `${prefix}_${MAP_IS_INITIALIZING}`:
             return {...state};
 
-        case FACEBOOK_LOGOUT_COMPLETE:
+        case `${prefix}_${FACEBOOK_LOGOUT_COMPLETE}`:
             return INITIAL_STATE;
 
         default:
             return state;
     }
-}
\ No newline at end of file
+};
+
+export default combineReducers({
+    add: formReducer(site_form_type.ADD),
+    edit: formReducer(site_form_type.EDIT)
+});
```

Comment:
Here's the trick that makes it all work. We combineReducers and create two identical ones that only differ by their invocation of either 'add' or 'edit'.